fbw-a32nx/src/systems/fmgc/src/flightplanning/FlightPlanService.ts

```diff
@@ -94,6 +93,22 @@ export class FlightPlanService<P extends FlightPlanPerformanceData = FlightPlanP
     return this.flightPlanManager.has(FlightPlanIndex.Uplink);
   }
 
+  async secondaryInit(index: number) {
+    if (this.flightPlanManager.has(FlightPlanIndex.FirstSecondary + index - 1)) {
```

Comment:
Should we log an error if we try to create a secondary plan when it already exists?

Reply:
> Maybe I should just remove the return and let it throw (and handle this better at the call site)?

I like this idea actually. We can handle the promise rejection in the caller.

Reply:
is this gonna throw an internal error scratchpad?